tests/lib/rust/time/src/lib.rs

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) 2024 Linaro LTD
+// SPDX-License-Identifier: Apache-2.0
+
+#![no_std]
+
+use core::ffi::{
+    c_char,
+    CStr,
+};
+
+use zephyr::printkln;
+use zephyr::time::{Duration, Instant, Tick, Timeout};
+use zephyr::raw::k_timeout_t;
+
+#[no_mangle]
+extern "C" fn rust_main() {
+    printkln!("Tick frequency: {}", zephyr::time::SYS_FREQUENCY);
+    // hack: sleep a bit to see if the twister failure can be earlier.
+    // zephyr::time::sleep(Duration::millis_at_least(1));
```

Comment:
pointing this out now just in case it gets forgotten

Reply:
So this was a stack overflow when twister would run the test, but not when run directly. Well, the stack always overflowed, but the test would pass if there was not tick interrupt during the test.

It seems that HW stack protection can be enabled anywhere, but is only implemented on some platforms.